src/modules/WebglDistort/Module.js

```diff
@@ -190,35 +190,61 @@ module.exports = function DoNothing(options, UI) {
     var parseDistortCoordinates = require('../../util/parseDistortCoordinates.js');
     var cornerCoordinates = parseDistortCoordinates(options);
 
-    if (!options.inBrowser) {
-      // this.output = input;
-      // callback();
-      require('../_nomodule/gl-context')(input, callback, step, options);
+    function extraManipulation(pixels, setRenderState, generateOutput) {
+      setRenderState(false);
+      if (!options.inBrowser) {
+        // this.output = input;
+        // callback();
+        require('../_nomodule/gl-context')(input, callback, step, options);
+      }
+      else {
+        var image = document.createElement('img');
+        
+        const getDataUri = require('../../util/getDataUri');
+        getDataUri(pixels, input.format).then(dataUri => {
```

Comment:
Why are we using getDataUri again, wouldn't this problem will be solved by #1571?

Reply:
We need to use it as this module requires the source of the pixels for its canvas element unlike the other modules in #1571